### ISSUE TITLE
[FP] Substring comparison flaw: multibytes chars

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1204]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         |  Php Inspections (EA Ultimate), 2.0.13|

Current behaviour
--
For the following code:
```php
if(substr($a, 0 , 8 ) == 'aaaéaaa'){
```
PHP EA Ultimate suggest to change the value from 8 to 7 to match with the compared string. However, the mb_substr was not used and the string contains a multibyte char (the "é" is in fact two bytes and count for 2 in substr). 

Here is a demo:
[3v4l demo](https://3v4l.org/U1X5c)

Expected behaviour
--
Take into account multi bytes chars and either suggest the correct number or don't trigger this inspection with such strings.
It may be relevant to suggest the use mb_substr and keep the correct number of chars.

**Comment by kalessil** [comment]
For myself: for `substr` filter all `\p{ASCII}` and add the resulted string length to expected one

**Comment by kalessil** [comment]
Fixed!
